### ISSUE TITLE
Quiz-box size

### ISSUE THREAD

**Issue by danm0nster** [nielsthl/QaDiL#7]
All quiz boxes seem to have the same size irrespective of the content. Sometimes it would be nice to have smaller boxes (if the options are all quite short, like a single number) and sometimes it would be nice to have slightly larger boxes (e.g. if the options are images).

Here is an example:
<img width="972" alt="Screenshot 2021-08-20 at 11 49 03" src="https://user-images.githubusercontent.com/16464590/130215164-4597534c-d0c4-431c-9e82-7e906dce7d2c.png">

Is there a good way to use images in quizzes?

**Comment by nielsthl** [comment]
The code that handles quizzes is in `qadil/Interactive.py`. It uses Bootstrap and (unfortunately) the `html` is hard wired into the `python` code as
```
<div class="col-xs-5 quizanswer" datav="{v}">{a}</div>
```
We should think of a more elegant design definitely using `jinja2` templates.

**Comment by danm0nster** [comment]
The reason that the images are cropped is that the height of the answer boxes is hard coded in `css/iLaTeXen.css`.

Here is the definition:
```
.quizanswer {
  border: solid 3px;
  border-radius: 8px;
  padding: 20px;
  height: 200px;
  margin-left: 10px;
  margin-right: 10px;
  margin-bottom: 10px;
  background-color: white; }
```

If remove the line setting the height it works fine, since then the row height will be set automatically by the boostrap grid system. See image below.

<img width="936" alt="Screenshot 2021-08-24 at 08 19 16" src="https://user-images.githubusercontent.com/16464590/130566285-6c4a6306-a035-4d54-86ec-4edaca52f0b5.png">

Perhaps a solution could be to have a default height that would be set in the HTML row tag rather than in CSS and then add an optional argument to the quiz environment like `\begin{quiz}[height=100]` for a smaller height, or `\begin{quiz}[height=auto]` for autoscaling (i.e. leave out the `height=` attribute in the HTML output).

**Comment by nielsthl** [comment]
Good idea. I removed `height` from my `css` and got some uneven boxes:
![quizcss](https://user-images.githubusercontent.com/23010769/130608429-9a18af59-bb18-496c-85b3-1f34b4962b1e.png)

An optional parameter could be a good addition. I was wondering if we could adjust the size of the boxes programmatically and define that uniform height to be the maximal height of a box plus epsilon.

**Comment by danm0nster** [comment]
max(height) + ε would be great, and I think we can get bootstrap to do that for us. See [this stackoverflow question](https://stackoverflow.com/questions/19695784/how-can-i-make-bootstrap-columns-all-the-same-height) and especially the answer called **Update 2021**. 

QaDiL uses bootstrap 3, but I could not seem to get the version 3 solution to work.

But another suggestion works, viz., adding this to the CSS:

```
  .row {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display:         flex;
    flex-wrap: wrap;
  }
  .row > [class*='col-'] {
    display: flex;
    flex-direction: column;
  }
```

Before:
<img width="884" alt="Screenshot 2021-08-25 at 12 47 13" src="https://user-images.githubusercontent.com/16464590/130777424-0771560e-3fe0-44d4-a5d2-3851199e8357.png">

After:
<img width="949" alt="Screenshot 2021-08-25 at 12 47 45" src="https://user-images.githubusercontent.com/16464590/130777567-394e5b98-1b30-49c7-beba-769f91e8f9d6.png">

I can commit this change or make a pull request. I haven't checked whether anything else is affected.

**Comment by danm0nster** [comment]
I spotted one side-effect. What used to take up one line suddenly splits up into three.

Before:
<img width="908" alt="Screenshot 2021-08-25 at 13 00 33" src="https://user-images.githubusercontent.com/16464590/130779246-875da49a-00fd-4a07-8a4f-a916d1592035.png">

After:
<img width="891" alt="Screenshot 2021-08-25 at 13 01 00" src="https://user-images.githubusercontent.com/16464590/130779303-1148ff1d-0b0a-4d3a-bf61-56add614cb7e.png">

Maybe upgrading to a newer bootstrap version is better?

**Comment by danm0nster** [comment]
Removing this part from the CSS file seems to get rid of the side effect.
```
  .row > [class*='col-'] {
    display: flex;
    flex-direction: column;
  }
```

**Comment by danm0nster** [comment]
Another side-effect is that boxes in the paraquiz environment change their vertical alignment. And the new paragraph (empty line in the TeX input file) seems to get gobbled up.

Before:

<img width="1038" alt="Screenshot 2021-08-25 at 15 54 58" src="https://user-images.githubusercontent.com/16464590/130803757-730f3c87-6901-4310-a9cc-baf3d61c142a.png">

After:

<img width="993" alt="Screenshot 2021-08-25 at 15 57 58" src="https://user-images.githubusercontent.com/16464590/130804305-19c85371-ed6a-442a-a942-327856f93d2f.png">


Maybe we just need to find the right magic CSS incantation.

**Comment by FunByJohn** [comment]
I think I fixed the issues.

Here is a [live demo](http://funbyjohn.com/QaDiLDemos/QuizBoxSizeIssue/intro.html).


Here are some other issuses that the code
``` 
.row {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display:         flex;
    flex-wrap: wrap;
  }
```
introduced:

<img width="507" alt="Screen Shot 2021-11-22 at 14 18 01" src="https://user-images.githubusercontent.com/3533905/142869791-8dda33e4-3ead-4705-aed9-3cbafd58c4ae.png">
^ Here the "X" and "Y" are placed badly

<img width="939" alt="Screen Shot 2021-11-22 at 14 17 51" src="https://user-images.githubusercontent.com/3533905/142869795-fffc8c46-9e96-48d0-b974-f98348c654c8.png">
^ It also messes with the orderquiz environment, and note how the math in the text looks bad here


My solution was to instead use the following CSS:
```
.quizboxes {
   display: -webkit-box;
   display: -webkit-flex;
   display: -ms-flexbox;
   display:         flex;
   flex-wrap: wrap;
}
```
and then in `qadil/Interactive.py` I change line 79 from
```
            returnstr += '<div class="row">'
```
to
```
            returnstr += '<div class="row quizboxes">'
```

The reason I went with this solution is that the `row` class is ubiquitous on the website, and it's only the actual quiz boxes in the `quiz` environment that we want to use the `flex` display style on.
